pkg/daemons/agent/agent.go

```diff
@@ -131,7 +131,14 @@ func startKubelet(cfg *config.Agent) error {
 	if err != nil || defaultIP.String() != cfg.NodeIP {
 		argsMap["node-ip"] = cfg.NodeIP
 	}
-	kubeletRoot, runtimeRoot, hasCFS, hasPIDs := CheckCgroups()
+	kubeletRoot, runtimeRoot, hasCFS, hasPIDs, cgroupsModeV2 := CheckCgroups()
+	if cgroupsModeV2 {
```

Comment:
&& only when the agent is running as the init

Reply:
What do you mean by "running as the init"?
Checking if it's PID 1 via `os.GetPid`?

Reply:
That should be fine, but didn't test by myself. Checking the length of /sys/fs/cgroup/cgroup.procs != 0 (as in the previous revision) would be also fine.

Reply:
:heavy_check_mark: